ako-crd-operator/internal/controller/routebackendextension_controller.go

```diff
@@ -0,0 +1,140 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/google/uuid"
+	akov1alpha1 "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-crd-operator/api/v1alpha1"
+	controllerutils "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-crd-operator/internal/utils"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
+
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-crd-operator/internal/cache"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-crd-operator/internal/constants"
+	avisession "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-crd-operator/internal/session"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+)
+
+// RouteBackendExtensionReconciler reconciles a RouteBackendExtension object
+type RouteBackendExtensionReconciler struct {
+	client.Client
+	AviClient     avisession.AviClientInterface
+	Scheme        *runtime.Scheme
+	Cache         cache.CacheOperation
+	Logger        *utils.AviLogger
+	EventRecorder record.EventRecorder
+	ClusterName   string
+}
+
+// +kubebuilder:rbac:groups=ako.vmware.com,resources=routebackendextensions,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ako.vmware.com,resources=routebackendextensions/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=ako.vmware.com,resources=routebackendextensions/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the RouteBackendExtension object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.21.0/pkg/reconcile
+func (r *RouteBackendExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Logger.WithValues("name", req.Name, "namespace", req.Namespace, "traceID", uuid.New().String())
+	ctx = utils.LoggerWithContext(ctx, log)
+	log.Debug("Reconciling RouteBackendExtension CRD")
+	defer log.Debug("Reconciled RouteBackendExtension CRD")
+	rbe := &akov1alpha1.RouteBackendExtension{}
+	err := r.Client.Get(ctx, req.NamespacedName, rbe)
+	if err != nil {
+		if k8serror.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		log.Error("Failed to get RouteBackendExtension CRD")
+		return ctrl.Result{}, err
+	}
+
+	if !rbe.DeletionTimestamp.IsZero() {
+		// The object is being deleted
+		r.EventRecorder.Event(rbe, corev1.EventTypeNormal, "Deleted", "RouteBackendExtension CRD deleted successfully from Avi Controller")
```

Comment:
What happens if the deletion fails or this is being referred in a httproute object ?

Reply:
We are not adding any finalizer to the object. We are not creating any controller side object, so there is no chance of deletion failing ( except underlying k8s env has issues). 
HTTPRoute has to handle the deletion event and need to do processing as per that. We don;t require to fail deletion because it is being referred in HTTPRoute.